allennlp/tests/modules/seq2seq_decoders/auto_regressive_seq_decoder_test.py

```diff
@@ -0,0 +1,90 @@
+import torch
+import pytest
+
+from allennlp.common.testing import AllenNlpTestCase
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.modules import Embedding
+from allennlp.modules.seq2seq_decoders import AutoRegressiveSeqDecoder
+from allennlp.modules.seq2seq_decoders import StackedSelfAttentionDecoderNet
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.util import START_SYMBOL, END_SYMBOL
+
+
+def create_vocab_and_decoder_net(decoder_inout_dim):
+    vocab = Vocabulary()
+    vocab.add_tokens_to_namespace(["A", "B", START_SYMBOL, END_SYMBOL])
+
+    decoder_net = StackedSelfAttentionDecoderNet(
+        decoding_dim=decoder_inout_dim,
+        target_embedding_dim=decoder_inout_dim,
+        feedforward_hidden_dim=20,
+        num_layers=2,
+        num_attention_heads=4,
+    )
+
+    return vocab, decoder_net
+
+
+class TestAutoRegressiveSeqDecoder(AllenNlpTestCase):
```

Comment:
Can you add a test for whether the metrics (`self._token_based_metric` and `self._tensor_based_metric`) get called when present?